akka/src/main/scala/cakesolutions/kafka/akka/KafkaConsumerActor.scala

```diff
@@ -10,22 +10,41 @@ import org.apache.kafka.common.serialization.Deserializer
 
 import scala.collection.JavaConversions._
 import scala.concurrent.duration._
-import scala.reflect.runtime.universe.{TypeTag, typeTag}
+import scala.reflect.runtime.universe.TypeTag
 
+/**
+  * An actor that wraps [[KafkaConsumerActor]].
+  *
+  * The actor pulls batches of messages from Kafka for all subscribed partitions,
+  * and forwards them to the supplied Akka actor reference in [[ConsumerRecords]] format.
+  *
+  * Before the actor continues pulling more data from Kafka,
+  * the receiver of the data must confirm the batches by sending back a [[KafkaConsumerActor.Confirm]] message
+  * that contains the offsets from the received batch.
+  * This mechanism allows the receiver to control the maximum rate of messages it will receive.
+  * By including the received offsets in the confirmation message,
+  * we avoid accidentally confirming batches that have not been fully processed yet.
```

Comment:
I note that `KafkaConsumer.Conf.apply(keyDeserializer: Deserializer[K]...)` is defined with this default: `enableAutoCommit: Boolean = true`.

So I think if someone creates a consumer conf using that function and doesn't set auto-commit to off, the underlying consumer will be automatically committing offsets on its own, which then makes the point about not accidentally confirming batches not quite true.

Am I mistaken somehow? If not, perhaps we should improve the documentation around this (but where?), or change `enableAutoCommit` to default to false. Or something else?